providers/file/file.go

```diff
@@ -48,7 +49,7 @@ func (f *File) Watch(cb func(event interface{}, err error)) error {
 	// the whole parent directory to pick up all events such as symlink changes.
 	fDir, _ := filepath.Split(f.path)
 
-	w, err := fsnotify.NewWatcher()
+	f.w, err = fsnotify.NewWatcher()
```

Comment:
This might end up spawning a new goroutine for each .Watch call? Also, since there can be only one path, it won't help calling it on the same path. Let's return an error if watcher (`if f.w != nil {return newerr}`) already exists.

Reply:
added a condition at the beginning of the function.